### ISSUE TITLE
Main window list columns cannot be managed individually

### ISSUE THREAD

**Issue by ineuw** [autokey/autokey#699]
## Classification:
Enhancement, UI/Usability

## Reproducibility:
Always

## AutoKey version:
0.96.0

## Used GUI:
Gtk

## Installed via:
pip3

## Linux distribution:
Linux Mint Cinnamon 20.3
Cinnamon 5.2.7

## Summary:
The left panel of the main window lists three columns but it's impossible adjust their widths and sort them as well. 

## Expected result:
Wanted to sort and change the width of the abbr.  and hotkey columns individually

## Actual result:
Cannot be adjusted

## Screenshot (if applicable):
![220614_frozen columns](https://user-images.githubusercontent.com/10367016/173507327-339e35da-abbb-4adc-ae2c-3a31c77e549d.jpg)

**Comment by josephj11** [comment]
I didn't get this at first, but being able to sort by abbreviation and hotkey would be very useful for managing/avoiding conflicts and deciding what to use for a new entry.

The use of window filters complicates this a bit because it can eliminate potential conflicts, but this isn't exposed in the panel. Fully supporting this might be difficult, but it might help to colorize or otherwise highlight entries which have window filters set.
.
A (sortable) windows filters column might be useful in this regard. It could be hidden by default if you don't want to sacrifice the screen space it would consume. This might make it useful to be able to sort by hotkey or abbreviation as a primary key with window filters as a secondary key. Alternatively, the window filter could display in a tooltip on hover.

Being able to "print"/export a sorted list could be useful as well - especially for users with large numbers of entries.

Screen real estate is always limited, so being able to resize the columns is not a bad idea either.
Some thought should be given to field values that are larger than the displayed field width (scrollable or wrapped fields).

Sorry for complicating this request, but it got me thinking. :)

**Comment by Elliria** [comment]
I'll join in with a +1 on the ability to rearrange and resize those columns. This is especially important when you don't open AutoKey in full-screen mode. I actually make mine bigger than I'd like it to be so that the columns show.

**Comment by ineuw** [comment]
Sorting the individual columns are far less important than the ability to adjust the column width of "Abbr".  I am never sure if I see everything typed.
P.S: Elliria's suggestion of rearranging the columns with sorting of the first column only, might be a solution.

**Comment by Elliria** [comment]
I can't take credit for that one. It was Joe's idea. I'm all for it, though. Being able to choose how to sort things gives us power and you never know which kind of power you might want at any given moment.

**Comment by ineuw** [comment]
I spent the last days cleaning up my Autokey scripts. 

I use keyboard activated scripts only. The origin of the scripts was Autohotkey in WIndows, which kept the key combinations in a single Autohotkey.ahk file, and after years of use it's the key combination is what I remember. The logic used for the key assignment was the most often used scripts were assigned key combinations I could activate by my left hand.

The attached image shows the filename starts with the key strokes, then if the script is all python (text formatting) or Wikisource  wiki language encapsulated using python, and finally if it's valid in both English and French Wikisource. 

If the column order could be rearranged, it would eliminate the need to add the keys to the name 
![Screenshot from 2022-07-13 10-33-30](https://user-images.githubusercontent.com/10367016/178861321-92969d46-9818-4ecb-87e2-693cb912b0b4.jpg)

**Comment by josephj11** [comment]
@ineuw That looks pretty extensive. Did you manage to "sell" it to any other Wikipedia editors?

**Comment by ineuw** [comment]
Posted some of my scripts in my workspace, but I had no feedback.

**Comment by josephj11** [comment]
Most people don't realize how important feedback (positive or negative) is until they create something and put it out there and then ... crickets.

**Comment by ineuw** [comment]
True, but there are thousands of superb programmers and developers on Wikipedia, truly nomadic and roaming around and contribute, and my scripts are preceded by numerous similar scripts, mostly in JavaScript, though Python is used more extensively than I thought, which in itself shows my ignorance.

**Comment by mapsedge** [comment]
Absolutely support sorting by column and setting column widths. This sort of functionality is so common in applications that I'm surprised it isn't there already.

**Comment by josephj11** [comment]
@mapsedge The bottleneck is our very limited developer resources. Nice, but not essential things can wait a long time.

**Comment by mapsedge** [comment]
I very much wish this was an area of development that I knew something about. My last desktop project was written in Visual Basic 6.0. I'm all PHP/web now. I'm game to try, but not sure how helpful I'd be, at least for a while.






---- On Mon, 22 May 2023 13:44:39 -0500 Joe ***@***.***> wrote ---





https://github.com/mapsedge The bottleneck is our very limited developer resources. Nice, but not essential things can wait a long time.

**Comment by josephj11** [comment]
@mapsedge Thanks for your interest!

We (and most projects) usually need all sorts of help, not just coding (although that's probably the scarcest resource). Here are a few things:

* Documentation, example scripts, workflow ideas, [wiki](https://github.com/autokey/autokey/wiki) content
* Testing new things (and betas when we have them) and recreating issues in other distros...
* We could really use someone who knows their way around Git/GitHub/ CI/CD better than we do - which isn't that high a bar ATM.
* User [community](https://github.com/autokey/autokey/wiki/Community) support and general participation - when we have new ideas... there are very few users who give us feedback
* We don't have our own website and I'm not sure we need one, but ...
* Feature requests and bug reports - like this one
* Answering questions, etc. on places like stack exchange that let other users and developers know we exist
* Writing articles about AutoKey

Anything you want to do will be appreciated.

Before embarking on any major effort though, it's a good idea to discuss what you want to do on [Gitter](https://gitter.im/autokey/autokey) first to make sure it has a good chance of being accepted and to draw on our experience as to the best way to proceed...

**Comment by Elliria** [comment]
That's a great list and most of it has now been added to the [Contributing](https://github.com/autokey/autokey/wiki/Contributing) wiki page.
